cmd/kubeadm/app/phases/kubelet/kubelet.go

```diff
@@ -152,5 +180,51 @@ func createKubeletBaseConfigMapRBACRules(client clientset.Interface) error {
 				Name: kubeadmconstants.NodesGroup,
 			},
 		},
+	}); err != nil {
+		return err
+	}
+
+	return apiclient.CreateOrUpdateClusterRoleBinding(client, &rbac.ClusterRoleBinding{
```

Comment:
Everything here can be just switched for one more subject in the above rolebinding -- just add the subject here to the RoleBinding above and it should work. That should grant `kubeadm join` using the token the right to get the configmap, right?